plugins/lrclib_lyrics/__init__.py

```diff
@@ -0,0 +1,210 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 Giorgio Fontanive (twodoorcoupe)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+PLUGIN_NAME = "Lrclib Lyrics"
+PLUGIN_AUTHOR = "Giorgio Fontanive"
+PLUGIN_DESCRIPTION = """
+Fetches lyrics from lrclib.net
+
+Also allows to export lyrics to an .lrc file or import them from one.
+"""
+PLUGIN_VERSION = "0.1"
+PLUGIN_API_VERSIONS = ["2.12"]
+PLUGIN_LICENSE = "GPL-2.0"
+PLUGIN_LICENSE_URL = "https://www.gnu.org/licenses/gpl-2.0.html"
+
+import os
+import re
+from functools import partial
+
+from picard import config, log
+from picard.file import register_file_post_save_processor, register_file_post_addition_to_track_processor
+from picard.track import Track
+from picard.ui.itemviews import BaseAction, register_track_action
+from picard.ui.options import OptionsPage, register_options_page
+from picard.webservice import ratecontrol
+
+from .option_lrclib_lyrics import Ui_OptionLrclibLyrics
+
+
+URL = "https://lrclib.net/api/get"
+REQUESTS_DELAY = 100
+
+# Options
+ADD_LYRICS = "add_lyrics"
+ADD_SYNCEDLYRICS = "add_syncedlyrics"
+NEVER_REPLACE_LYRICS = "never_replace_lyrics"
+LRC_FILENAME = "exported_lrc_filename"
+EXPORT_LRC = "exported_lrc"
+NEVER_REPLACE_LRC = "never_replace_lrc"
+
+lyrics_cache = {}
+synced_lyrics_pattern = re.compile(r"(\[\d\d:\d\d\.\d\d\d]|<\d\d:\d\d\.\d\d\d>)")
+tags_pattern = re.compile(r"%(\w+)%")
+extra_file_variables = {
+    "filepath": lambda file: file,
+    "folderpath": lambda file: os.path.dirname(file),
+    "filename": lambda file: os.path.splitext(os.path.basename(file))[0],
+    "filename_ext": lambda file: os.path.basename(file),
+    "directory": lambda file: os.path.basename(os.path.dirname(file))
+}
+
+
+def get_lyrics(track, file):
+    album = track.album
+    metadata = file.metadata
+    if not (config.setting[ADD_LYRICS] or config.setting[ADD_SYNCEDLYRICS]):
+        return
+    if not (metadata.get("title") and metadata.get("artist")):
+        log.debug(f"Skipping fetching lyrics for track in {album} as both title and artist are required")
+        return
+    if config.setting[NEVER_REPLACE_LYRICS] and metadata.get("lyrics"):
+        log.debug(f"Skipping fetching lyrics for {metadata["title"]} as lyrics are already embedded")
```

Comment:
```suggestion
        log.debug(f"Skipping fetching lyrics for {metadata['title]} as lyrics are already embedded")
```

Nested inputs need to use single quotes.